src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -39,6 +39,69 @@ static bool arg_enabled = true;
 static bool arg_root_enabled = true;
 static int arg_root_rw = -1;
 
+static int open_parent_devno(dev_t devnum, int *ret) {
```

Comment:
call the param ret_fd, maybe? Why "devno" in the name btw? the fact that this uses devnos internally is prett ymuch hidden, no? maybe call call open_parent_block_device? also, why not just return the fd as return value and ENODEV or ENOTBLK when the device is not suitable?

I mean, our usual coding sytle says we need to initialize return params in all cases of success, but this function returns 0 (which is success) but still doesn't initialize the return value. I think prettiest really would be to directly return the fd without a return param

Reply:
oh, i see now that the function isn't new...

Reply:
I changed the function name and return parameter name, but didn't change the function to return the fd. It turned out to be really inconvenient, because the function ignores most errors, but not all, and it becomes messy to differentiate the two if 0 is not available to signal failure that should be ignored.